pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -107,14 +110,35 @@ func (c *Cluster) createStorageClassDeviceSetPVC(storageClassDeviceSetName strin
 		// The PVC is already present
 	} else if presentPVCsNum == 1 {
 		logger.Debugf("already present pvc %q", presentPVCs.Items[0].Name)
+		c.updatePVCIfChanged(pvc, &presentPVCs.Items[0])
+
 		// Updating with the new label
 		return &presentPVCs.Items[0], nil
 	}
 	// More than one PVC exists with same labelSelector
 	return nil, errors.Errorf("more than one PVCs exists with label %q, pvcs %q", pvcStorageClassDeviceSetPVCIdLabelSelector, presentPVCs)
 }
 
-func makeStorageClassDeviceSetPVC(storageClassDeviceSetName, pvcStorageClassDeviceSetPVCId string, setIndex int, pvcTemplate v1.PersistentVolumeClaim) (pvcs *v1.PersistentVolumeClaim) {
+func (c *Cluster) updatePVCIfChanged(desiredPVC *v1.PersistentVolumeClaim, currentPVC *v1.PersistentVolumeClaim) {
+	desiredSize, desiredOK := desiredPVC.Spec.Resources.Requests[v1.ResourceStorage]
+	currentSize, currentOK := currentPVC.Spec.Resources.Requests[v1.ResourceStorage]
+	if !desiredOK || !currentOK {
+		logger.Debugf("desired or current size are not specified for pvc %q", currentPVC.Name)
+		return
+	}
+	if desiredSize.Value() > currentSize.Value() {
```

Comment:
Should we check if `desiredSize.Value() < currentSize.Value()` and print a message explaining it's unsupported?

Reply:
Agreed, will add a log message in that case.